pkg/kvstore/etcd.go

```diff
@@ -580,7 +580,7 @@ reList:
 					}
 
 					// Retrieve latest revision
-					lastRev = 0
+					lastRev = 0 // ineffassign: ignore
```

Comment:
Could we just move this line and/or the declaration of lastRev to the beginning of the `reList` loop? That's the only way we would end up with lastRev=0 anyway... I might have missed something, this code is incredibly difficult to follow.
While supporting ignore is handy, it defeats the point of using ineffassign. The tool is telling us that we've done something unintended in the language so our code is very likely incorrect.

Reply:
You're right, I will try that and submit a separate PR if moving that assignment to the beginning of `reList` will work

Reply:
Thanks. To be clear, I'm not opposed to this PR being merged, but in the long run I'm worried the ignore will allow some bad behaviour on our part ("Oh, I just need to merge this to make this deadline" and then forgetting about it).